### PR TITLE
Move methods which are entities' responsibility into entities

### PR DESCRIPTION
**Pull request by TomasHofman**
Aphrodite#getIssue(PullRequest) -> PullRequest#findIssueURL()
Aphrodite#getRelatedIssues(PullRequest) -> PullRequest#findRelatedIssuesURL()
Aphrodite#getUpstreamIssue(PullRequest) -> PullRequest#findUpstreamIssueURL()
Aphrodite#getUpstreamPullRequest(PullRequest) -> PullRequest#findUpstreamPullRequestURL()
Aphrodite#getPullRequestUpgrade(PullRequest) -> PullRequest#findPullRequestUpgradeURL()
Aphrodite#isUpstreamRequired(PullRequest) -> PullRequest#isUpstreamRequired()
Aphrodite#hasUpgrade(PullRequest) -> PullRequest#hasUpgrade()

Aphrodite#extractURLs(String, Pattern, boolean) -> URLUtils#extractURLs(String, Pattern, boolean)

Plus some javadoc fixes.

### CODE REVIEW

**Comment by TomasHofman** [comment]
@wolfc @baranowb I guess it's safe to merge this one? It's mostly clean up, method version that return entities instead of URLs can be added later.

File: common/src/main/java/org/jboss/set/aphrodite/Aphrodite.java [hunk 1]
```diff
@@ -236,184 +221,6 @@ public Issue getIssue(URL url) throws NotFoundException {
...
-     * @param pullRequest - PR which will be interrogated.
-     * @return
-     *         <ul>
-     *         <li>null</li> - if no issue is linked in PR
-     *         <li>Issue</li> - if there is linked issue that can be fetched
-     *         </ul>
-     * @throws NotFoundException - if there is linked issue but either no tracker or issue does not exist in tracker
-     * @throws MalformedURLException
-     */
-    public Issue getIssue(final PullRequest pullRequest) throws NotFoundException, MalformedURLException {
```
Comment (wolfc) [inline common/src/main/java/org/jboss/set/aphrodite/Aphrodite.java]: Can we not deprecate these public methods first?

File: domain/src/main/java/org/jboss/set/aphrodite/domain/util/URLUtils.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package org.jboss.set.aphrodite.domain.util;
```
Comment (wolfc) [inline domain/src/main/java/org/jboss/set/aphrodite/domain/util/URLUtils.java]: Maybe internal package?
Not sure whether there is another way to ensure it does not get exposed.

File: domain/src/main/java/org/jboss/set/aphrodite/domain/util/URLUtils.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    /**
+     * Extracts URLs matching given pattern from given String.
+     *
+     * @param source String to extract URLs from.
+     * @param initialMatchPattern Pattern that URLs must match.
+     * @param multiple If false, return the first URL only, if true, return all.
+     * @return URLs matching <code>initialMatchPattern</code>
+     */
+    public static String[] extractURLs(final String source, final Pattern initialMatchPattern, final boolean multiple) {
```
Comment (wolfc) [inline domain/src/main/java/org/jboss/set/aphrodite/domain/util/URLUtils.java]: Missing test case.

File: common/src/main/java/org/jboss/set/aphrodite/Aphrodite.java (hunk 1 above)
Comment (TomasHofman) [reply common/src/main/java/org/jboss/set/aphrodite/Aphrodite.java]: OK, deprecated.

File: domain/src/main/java/org/jboss/set/aphrodite/domain/util/URLUtils.java (hunk 2 above)
Comment (TomasHofman) [reply domain/src/main/java/org/jboss/set/aphrodite/domain/util/URLUtils.java]: I renamed it to "internal". We would have to use something like jboss-modules if we wanted to enforce it.

File: domain/src/main/java/org/jboss/set/aphrodite/domain/util/URLUtils.java (hunk 3 above)
Comment (TomasHofman) [reply domain/src/main/java/org/jboss/set/aphrodite/domain/util/URLUtils.java]: Added. I modified the original reg exp a little, now it supports credentials and IPv4 addresses too.

**Pull request merged**